### PR TITLE
Separate Transformers::Article into Scraper and Trainer

### PR DESCRIPTION
**Pull request by richardwu**
Key note:
- Abstracted common methods (everything but constructor) into a `Transformers::Article` module
- Created a `Transformers::ScraperArticle` and `Transformers::TrainerArticle` class

Would inheritance be better here? 

cc: @jules2689

### CODE REVIEW

File: lib/news_scraper/transformers/article.rb [hunk 1]
```diff
@@ -5,50 +5,35 @@
...
 
       private
 
-      def scrape_details
-        @scrape_details ||= @scrape_patterns['domains'][@root_domain]
-      end
-
       def transformed_response
-        @scrape_patterns['data_types'].each_with_object({}) do |data_type, response|
+        Constants::SCRAPE_PATTERNS['data_types'].each_with_object({}) do |data_type, response|
```
Comment (jules2689) [inline lib/news_scraper/transformers/article.rb:43]: shouldn't this stay as before?

**Comment by jules2689** [comment]
Inheritance probably makes sense, also you have merge conflicts

File: lib/news_scraper/transformers/article.rb (hunk 1 above)
Comment (richardwu) [reply lib/news_scraper/transformers/article.rb:43]: There's no point in initializing an instance variable for scrape_patterns.

Comment (jules2689) [reply lib/news_scraper/transformers/article.rb:43]: Oh, for data types, I suppose. Other ones will make a difference

**Comment by richardwu** [comment]
re-ping

**Comment by jules2689** [comment]
LGTM

**Pull request merged**